sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py

```diff
@@ -1053,10 +1052,10 @@ async def test_async_queue_cancel_scheduled_messages(self, servicebus_namespace_
                 async with sb_client.get_queue_sender(servicebus_queue.name) as sender:
                     message_a = Message("Test scheduled message")
                     message_b = Message("Test scheduled message")
-                    tokens = await sender.schedule_messages([message_a, message_b], enqueue_time)
+                    tokens = await sender.schedule_messages([message_a, message_b], enqueue_time, timeout=0)
                     assert len(tokens) == 2
 
-                    await sender.cancel_scheduled_messages(tokens)
+                    await sender.cancel_scheduled_messages(tokens, timeout=None)
```

Comment:
We put the timeouts in; and we have the test at the top for happy path, but do we ever test that a timeout actually hits in any way?  (Don't need to do it for all methods, just choose one and monkeypatch the internals so that the timeout loop is hit.  Bit janky but I don't feel too bad if we hijack something at a contract boundary that's not likely to change as an implementation detail.

Reply:
tested added, thanks for the call out of adding tests which helps find a bug in uamqp :)